### ISSUE TITLE
Surprising nullable warning after null-conditional property access

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#39858]
**Version Used**: 
16.4.0 Preview 5.0

**Steps to Reproduce**:
```c#
#nullable enable
using System;
public class C
{
    public void M(string value)
    {
        Console.WriteLine(value.Length);
        Console.WriteLine(value?.Length);
        Console.WriteLine(value.Length); // CS8602 for value
    }
}
```

**Expected Behavior**:
I would have expected a warning for the middle line, because the null-conditional access is superflous.

**Actual Behavior**:
Instead in the third statement, the compiler decides that the non-nullable `value` could be null. Hovering `value` in VS doesn't show any nullable annotation.

**Comment by gafter** [comment]
When you use a null-conditional operator, you are telling the compiler that the value on the left might be null.  That's why you get a warning on the next line.  This is by design.

**Comment by Tragetaschen** [comment]
But shouldn't there be a warning on the second line then, the `?.` on a non-nullable type?

**Comment by gafter** [comment]
The purpose of the feature is to warn you when you dereference a possibly-null value.  It is not the aim of the feature to tell you about unnecessary null tests, as the compiler does not know if the incoming value might actually contain null despite the declared type.
